src/config/enterprise.rs

```diff
@@ -0,0 +1,426 @@
+use std::{
+    env,
+    fmt::{Display, Formatter},
+};
+
+use http::Request;
+use hyper::{header::LOCATION, Body, StatusCode};
+use serde::{Deserialize, Serialize};
+use tokio::{
+    select,
+    time::{sleep, Duration},
+};
+use url::{ParseError, Url};
+
+use super::{
+    load_source_from_paths, process_paths, ComponentKey, Config, ConfigPath, OutputId, SinkOuter,
+    SourceOuter,
+};
+use crate::{
+    common::datadog::get_api_base_endpoint,
+    http::{HttpClient, HttpError},
+    signal::{SignalRx, SignalTo},
+    sinks::datadog::metrics::DatadogMetricsConfig,
+    sources::{host_metrics::HostMetricsConfig, internal_metrics::InternalMetricsConfig},
+};
+use vector_core::config::proxy::ProxyConfig;
+
+static HOST_METRICS_KEY: &str = "#datadog_host_metrics";
+static INTERNAL_METRICS_KEY: &str = "#datadog_internal_metrics";
+static DATADOG_METRICS_KEY: &str = "#datadog_metrics";
```

Comment:
Yes, this could certainly clash.

Historically, a `#` prefix was chosen because the `[<type>.<id>]` syntax of TOML prevents hashes being used. Practically, this should prevent hashes being used in the wild... although other formats and definitions allow it.

At one stage, an idea was floated for a UUID... but this then makes the component ID non-deterministic, which it probably ought to be.

Ideally, the longer-term solution would be to:

1. Have a syntax that allows for 'unspeakable' component IDs that definitely won't clash with public names.
2. Mark a component as private internally, so we can mark it out in the API for use-cases like top/tap.

It may be relevant to @tobz's work per #11634 to identify valid component IDs so that we can use internally invalid variants, but there is also likely to be work needed internally to sidestep these requirements which also hasn't been done (at least not by me.)

Reply:
Gotcha. I assume it's not a big deal for now since users won't yet be using `[enterprise]`.